app/services/outliers/calculate_centroid.rb

```diff
@@ -0,0 +1,22 @@
+class Outliers::CalculateCentroid
+  def call(measurements)
+    lng_lat = sum_coordinates(measurements)
+    average(lng_lat, BigDecimal.new(measurements.size))
+  end
+
+  private
+
+  def sum_coordinates(measurements)
+    measurements.reduce([zero, zero]) do |(lng, lat), x|
+      [lng + x.longitude, lat + x.latitude]
+    end
+  end
+
+  def average(lng_lat, amount)
+    lng_lat.map { |x| x / amount }
```

Comment:
Is it 100% sure that amount will never be 0? If it's not an issue in this case then ignore this comment :)

And it's just a detail, but it might be better to call it number. Amount is for uncountable stuff. That's how I remember :D https://pl.pinterest.com/pin/650910952367202906/

Reply:
Good catch! I've added a guard in the calling method (https://github.com/HabitatMap/AirCasting/pull/52/commits/a5f136f028d6f93f848fa11c3e2e34c2a1b730a3)